### PR TITLE
Add Tesla Fleet integration

### PR DESCRIPTION
**Pull request by Bre77**
## Proposed change

Add Tesla Fleet API documentation

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [x] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [x] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/122019
- Link to parent pull request in the Brands repository: https://github.com/home-assistant/brands/pull/5652
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

## Summary by CodeRabbit

- **Documentation**
  - Added detailed documentation for integrating the Tesla Fleet API with Home Assistant. This includes prerequisites, available entities (vehicles, energy sites, wall connectors), and methods to manage vehicle sleep. Users can now utilize various sensors to monitor Tesla vehicles and energy sites, including battery level, charge status, and temperatures.

### CODE REVIEW

File: source/_integrations/tesla_fleet.markdown [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+ha_config_flow: true
+ha_codeowners:
+  - '@Bre77'
+ha_domain: tesla_fleet
+ha_platforms:
+  - sensor
+ha_integration_type: integration
+---
+
+The Tesla Fleet API integration exposes various sensors from Tesla vehicles and energy sites.
```
Comment (c0ffeeca7) [inline source/_integrations/tesla_fleet.markdown]:
```suggestion
The Tesla Fleet API {% term integration %} exposes various sensors from Tesla vehicles and energy sites.
```

File: source/_integrations/tesla_fleet.markdown [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+{% include integrations/config_flow.md %}
+
+## Scopes
+
+When connecting your Tesla account to Home Assistant, you **must** select the `Vehicle Information` or `Energy Product Information` scope. It is recommended you select all scopes for full functionality.
+
+## Rate limits
+
+Tesla restricts open-source integrations to the ["Discovery tier"](https://developer.tesla.com/docs/fleet-api#membership-levels) which only allows for 200 vehicle data requests per day. The integration will stop polling for vehicle updates when it hits this limit, and will wait the appropriate time specified in the Retry-After header.
```
Comment (c0ffeeca7) [inline source/_integrations/tesla_fleet.markdown]:
```suggestion
Tesla restricts open-source integrations to the ["Discovery tier"](https://developer.tesla.com/docs/fleet-api#membership-levels) which only allows for 200 vehicle data requests per day. The integration will stop polling for vehicle updates when it hits this limit, and will wait the appropriate time specified in the **Retry-After** header.
```

File: source/_integrations/tesla_fleet.markdown [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+|Sensor|Scheduled charging mode|No|
+|Sensor|Scheduled charging start time|No|
+|Sensor|Scheduled departure time|No|
+|Sensor|Shift state|No|
+|Sensor|Speed|No|
+|Sensor|State of charge at arrival|No|
+|Sensor|Time at arrival|Yes|
+|Sensor|Time at full charge|Yes|
+|Sensor|Time to arrival|Yes|
+|Sensor|Time to arrival|Yes|
```
Comment (c0ffeeca7) [inline source/_integrations/tesla_fleet.markdown]: are these duplicates on purpose? what is the difference between them?

File: source/_integrations/tesla_fleet.markdown [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+|Sensor|Time to full charge|Yes|
+|Sensor|Time to full charge|Yes|
```
Comment (c0ffeeca7) [inline source/_integrations/tesla_fleet.markdown]: Are these duplicates on purpose? What is the difference?

File: source/_integrations/tesla_fleet.markdown [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+|Sensor|Tire pressure front left|No|
+|Sensor|Tire pressure front right|No|
+|Sensor|Tire pressure last measured front left|No|
+|Sensor|Tire pressure last measured front right|No|
+|Sensor|Tire pressure last measured rear left|No|
+|Sensor|Tire pressure last measured rear right|No|
+|Sensor|Tire pressure rear left|No|
+|Sensor|Tire pressure rear right|No|
+|Sensor|Traffic delay|No|
+|Sensor|Usable Battery level|No|
```
Comment (c0ffeeca7) [inline source/_integrations/tesla_fleet.markdown]:
```suggestion
|Sensor|Usable battery level|No|
```

**Review by c0ffeeca7 — CHANGES_REQUESTED** [review]

File: source/_integrations/tesla_fleet.markdown (hunk 3 above)
Comment (Bre77) [reply source/_integrations/tesla_fleet.markdown]: This was an oversight from the place I copied it from.

File: source/_integrations/tesla_fleet.markdown (hunk 4 above)
Comment (Bre77) [reply source/_integrations/tesla_fleet.markdown]: This was an oversight from the place I copied it from.

**Review by c0ffeeca7 — APPROVED** [review]
Thank you, @Bre77 :+1:

**Pull request merged**